syft/rekor/rekor_test.go

```diff
@@ -0,0 +1,223 @@
+package rekor
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/source"
+	"github.com/go-openapi/runtime"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/client/entries"
+	"github.com/sigstore/rekor/pkg/generated/client/index"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/stretchr/testify/assert"
+)
+
+type testCase struct {
+	name                string
+	inputFilePath       string // resolver and location are created from this
+	expectedOutput      []artifact.Relationship
+	uuidsToLogEntryFile map[string]string // tells the rekorClient mock what uuids and what log entries to return
+	httpClient          *http.Client
+	expectedErr         string
+	expectedLog         string
+}
+
+/*
+***************
+
+	Functions and types to complete the interfaces in client.Rekor. Most are unimplemented.
+
+****************
+*/
+
+type entriesMock struct {
```

Comment:
fyi

not sure if you'll find this useful but you can technically have indexMock and entriesMock be the same struct and then you only have to define all the interfaces for 1 structs. Since interfaces are a function of what's implemented..

And then you can typedef it to a struct if you want to...